lib/person_creator.rb

```diff
@@ -0,0 +1,72 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'classroom'
+
+# The personCreator class responsible for creating new Person objects
+
+class PersonCreator
+  def initialize(people)
+    @people = people
+  end
+
+  def create_student
+    name = read_person_name_from_user_input('Student')
+    age = read_person_age_from_user_input
+    parent_permission = read_parent_permission_from_user_input
+    classroom_label = read_classroom_label_from_user_input
+
+    create_new_student(name, age, parent_permission, classroom_label)
+    puts 'Student created successfully:'
+  rescue ArgumentError => e
+    puts "Error: #{e.message}"
+  end
+
+  def read_person_name_from_user_input(type)
+    print "#{type} name: "
+    gets.chomp
+  end
+
+  def read_person_age_from_user_input
+    print 'Age: '
+    gets.chomp.to_i
+  end
+
+  def read_parent_permission_from_user_input
+    print 'Has parent permission? (y/n): '
+    gets.chomp.downcase == 'y'
+  end
+
+  def read_classroom_label_from_user_input
+    print 'Classroom label: '
+    gets.chomp
+  end
+
+  def create_new_student(name, age, parent_permission, classroom_label)
+    student = Student.new(name, age, parent_permission: parent_permission,
```

Comment:
**Feedback for the `PersonCreator` Class:**

Great effort on the `PersonCreator` class! I've reviewed your code and found something that might be causing a little issue as shown in the image below 👇🏾 

<img width="296" alt="image" src="https://github.com/Nasirkhan294/school_library/assets/84156714/9950f7cf-2973-4224-b96b-3df84cef284b">

<br />

In your `create_new_student` method on line 45, it seems that the arguments are passed in the wrong order when creating a new `Student` object. The `Student` class's constructor takes the parameters in this order: `age`, `name="default"`, `parent_permission=true`, `classroom=nil`. However, in the `create_new_student` method, you are passing the arguments in a different order, which might be confusing.

This mismatch in parameter order could explain why, when you try to list people, the student's age appears in place of the name, and the name in place of the age.

To fix this, make sure you pass the arguments in the correct order when creating a new `Student` object. It should be:

```ruby
Student.new(age, name, parent_permission, classroom)
```

By following this order, your code should work as expected, and the information for each student will be displayed correctly.

Keep up the good work! If you have any questions or need further assistance, feel free to ask. Happy coding!